src/hooks/postupdate.ts

```diff
@@ -70,8 +69,8 @@ const hook: Hook.Update = async function (opts): Promise<void> {
       sfExec = path.join(dataDirBin, 'sf');
     }
 
-    writeFileSync(sfExec, readFileSync(sfdxExec, { encoding: 'utf-8' }).replace(/sfdx/g, 'sf').replace(/SFDX/g, 'SF'));
-    chmod('+x', dataDirBin, sfExec);
+    const sfExecContents = readFileSync(sfdxExec, { encoding: 'utf-8' }).replace(/sfdx/g, 'sf').replace(/SFDX/g, 'SF');
+    writeFileSync(sfExec, sfExecContents, { mode: 0o755 });
```

Comment:
I tested this a bunch and it works the same as the replaced code on MacOS.  I did not test on Linux but I'm assuming it's the same.  Windows would need testing, but I'm not confident (based on shelljs and nodejs docs) that the replaced code would work on Windows.

Reply:
from what I found on Windows only the write perms can be modified so this works for the `sf` executable.

https://nodejs.org/api/fs.html#fsopenpath-flags-mode-callback